### PR TITLE
AbstractTexture: Move Bind() method to Renderer

### PR DESCRIPTION
**Pull request by stenzek**
This makes state tracking simpler, and enables easier porting to command lists later on.

### CODE REVIEW

File: Source/Core/VideoBackends/OGL/OGLTexture.cpp [hunk 1]
```diff
@@ -120,17 +117,13 @@ OGLTexture::OGLTexture(const TextureConfig& tex_config) : AbstractTexture(tex_co
...
 }
 
 OGLTexture::~OGLTexture()
 {
   if (m_texId)
   {
-    for (auto& gtex : s_Textures)
-      if (gtex == m_texId)
-        gtex = 0;
+    g_renderer->UnbindTexture(this);
```
Comment (iwubcode) [inline Source/Core/VideoBackends/OGL/OGLTexture.cpp]: Isn't this handled by the base AbstractTexture class?

Comment (stenzek) [reply Source/Core/VideoBackends/OGL/OGLTexture.cpp]: Yep, good catch.

File: Source/Core/VideoBackends/D3D/Render.cpp [hunk 2]
```diff
@@ -733,11 +733,24 @@ void Renderer::SetDepthState(const DepthState& state)
   m_gx_state.zmode.hex = state.hex;
 }
 
+void Renderer::SetTexture(u32 index, const AbstractTexture* texture)
+{
+  D3D::stateman->SetTexture(
+      index,
+      texture ? static_cast<const DXTexture*>(texture)->GetRawTexIdentifier()->GetSRV() : nullptr);
```
Comment (degasus) [inline Source/Core/VideoBackends/D3D/Render.cpp]: I'm not familiar with out D3D helpers, but what shall happen if texture is nullptr? Should we call UnbindTexture instead?

**Comment by degasus** [comment]
LGTM

**Pull request merged**